operator/pkg/ciliumendpointbatch/endpointbatch.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointbatch
+
+import (
+	"context"
+	"time"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	capi_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	csv2 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2"
+	csv2a1 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// cebNamePrefix is the prefix name for added for CiliumEndpointBatch
+	// resource.
+	cebNamePrefix = "ceb"
+
+	// CEPs are queued in a CEB as FirstComeFirstServe, CEPs are packed in random order.
+	CEBatchingModeFcfs = "cebModeFcfs"
+)
+
+var (
+	// apiServerBackoff creates a new API Machinery backoff parameter.
+	apiServerBackoff = wait.Backoff{
+		// Return a exponential backoff configuration, return total duration of ~2 second.
+		// Example, 0s, 0.055s, 0.320s, 1.65s
+		// Maximum number of retries is 4
+		Steps:    4,
+		Duration: 50 * time.Millisecond,
+		Factor:   5.0,
+		Jitter:   0.1,
+	}
+)
+
+type CiliumEndpointBatchController struct {
+	// Cilium kubernetes clients to access V2 and V2alpha1 resources
+	clientV2   csv2.CiliumV2Interface
+	clientV2a1 csv2a1.CiliumV2alpha1Interface
+
+	// aggregator, is an util used to order the list of CEB that needs to reconciled with API server.
+	aggregator *aggregator
+
+	// reconciler is an util used to reconcile CiliumEndpointBatch changes.
+	reconciler *reconciler
+
+	// Manager is used to create and maintain local datastore. Manager watches for
+	// cilium endpoint changes and enqueue/dequeue the cilium endpoint changes in CEB.
+	// It maintains the desired state of CEBs in dataStore
+	Manager cebManager
+
+	// cebDeleteSyncTime is used track latency between CEBatchDeleteSyncPeriod. if the latency
+	// exceeds CEBatchDeleteSyncPeriod it would let reconcile all CEB's [create, update and delete]
+	// with k8s-apiServer.
+	cebDeleteSyncTime time.Duration
+
+	// ciliumEndpointStore is used to list of current active CEPs in a cluster. By, using this cache
```

Comment:
```suggestion
	// ciliumEndpointStore is a list of current active CEPs in a cluster. By, using this cache
```

Reply:
CiliumEndpointStore is an interface used to get the list of current active CEPs in a cluster.

Reply:
Okay, but "used to list of" is not a valid sentence. Maybe change to "used to get a list of"?

Reply:
Sure,  let me rephrase the comment.